mlrun/serving/remote.py

```diff
@@ -205,8 +215,10 @@ def do_event(self, event):
 
     def _generate_request(self, event, body):
         method = self.method or event.method or "POST"
-        headers = self.headers or {}
-
+        if self._headers_function_handler:
+            headers = self._headers_function_handler(body)
+        else:
+            headers = self.headers or {}
```

Comment:
Not a regression in this PR, but I'm pretty sure this should be
```suggestion
            headers = copy(self.headers) or {}
```
to avoid mutating `self.headers`.